add initial offset to kafka source (#4080)
knative/docs: docs/developer/eventing/sources/kafka-source/README.md

        name: event-display
   ```

## (Optional) Specify the initial offset

> By default the `KafkaSource` starts consuming from the `latest` offset in each partition. In case, you want to consume from the earliest offset, use this field to configure the same.

Reviewer: `to configure the same` does not sound right. What about `In case you want to consume from the earliest offset, set the initialOffset field to earliest`?

> By default the `KafkaSource` starts consuming from the `latest` offset in each partition. In case you want to consume from the earliest offset, set the initialOffset field to `earliest`.

Reviewer 2: ```suggestion
By default the `KafkaSource` starts consuming from the latest offset in each partition. If you want to consume from the earliest offset, set the `initialOffset` field to `earliest`, for example:
```

   ```yaml
   apiVersion: sources.knative.dev/v1beta1
   kind: KafkaSource
   metadata:
    name: kafka-source
   spec:
    consumerGroup: knative-group
    initialOffset: earliest
    bootstrapServers:
    - my-cluster-kafka-bootstrap.kafka:9092 # note the kafka namespace
    topics:
    - knative-demo-topic
    sink:
      ref:
        apiVersion: serving.knative.dev/v1
        kind: Service
        name: event-display
   ```

> *NOTE:* valid values for `initialOffset` is `earliest` or `latest`, any other value would result in a validation error. This field will be honored only if there are no prior committed offsets for that consumer group.

Reviewer 2: ```suggestion
!!! note
    The valid values for `initialOffset` are `earliest` or `latest`. Any other value results in a validation error. This field is honored only if there are no prior committed offsets for that consumer group.
```

> *NOTE:* valid values for `initialOffset` is `earliest` or `latest`, any other value would result in a validation error. Also this field will be honored, only if there are no prior committed offsets for that consumer group.

Reviewer: I would remove `Also` and the comma after `honored`.
